src/pallene/coder.lua

```diff
@@ -593,59 +593,99 @@ function Coder:lua_entry_point_definition(f_id)
 
     self.current_func = func
 
-    -- We unconditionally initialize the `K` userdata here, in case one of the tag checking tests
-    -- needs to use it. We don't bother to make this initialization conditional because in the case
-    -- that really matters (small leaf functions that don't use `K`) the C compiler can optimize this
-    -- read away after inlining the Pallene entry point.
-    local init_global_userdata = [[
+    -- 1) Adjust input arguments
+    --
+    -- Raises an error if the number of arguments is incorrect
+    -- Fills missing optional arguments with nil
+
+    local adjust_arguments
+    do
+        local min_nargs   = types.number_of_mandatory_args(arg_types)
+        local max_nargs   = #arg_types
+
+        local tmpl
+        if min_nargs == max_nargs then
+            tmpl = [[
+                int nargs = lua_gettop(L);
+                if (nargs != $min_nargs) {
+                    pallene_runtime_arity_error(L, $fname, $min_nargs, $max_nargs, nargs);
+                }
+            ]]
+        else
+            tmpl = [[
+                int nargs = lua_gettop(L);
+                if (nargs < $min_nargs || nargs > $max_nargs) {
+                    pallene_runtime_arity_error(L, $fname, $min_nargs, $max_nargs, nargs);
+                }
+                for (; nargs < $max_nargs; nargs++) {
+                    ${push_nil}
+                }
+            ]]
```

Comment:
If I am not wrong, Pallene does not support Variable Arguments. Thing is, variable arguments goes really well with the "Optional Argument" idea. Maybe we can have something like this instead: 

```
if nargs < min_args: 
    error

total_args = nargs - passed_arg
if total_args > 0:
    push_nils
else 
    pop_values
```

Or in case of more arguments than expected, instead of popping, we can pull them to vararg which can be accessed through `(...)` like Lua?

Reply:
Sorry, I didn't see the notification for your review so I already merged it...

Indeed, optional arguments does help us move towards varargs in the future! However, properly handling varargs will require many more changes than the one you suggested here. I decided it was probably best to deal with those only when we actually implement varargs.

Reply:
For example, varargs require us to reorder the stuff in the stack. When a Lua function receives a stack like

     [args] [varargs]
     ^stack base
     
 It has to change it into

    [nils] [varargs] [args]
                     ^stack base

so that the varargs are in the "negative" slots and do not interfere with the math for the local variable stack slots.